integration/test/Test/LegalHold.hs

```diff
@@ -274,3 +280,522 @@ testLHDeleteClientManually = do
+          mzero
+
+      void $ local (setTimeoutTo 90) do
+        awaitNotification bob bobc noValue isUserClientRemoveNotif
+          *> awaitNotification bob bobc noValue isUserLegalholdDisabledNotif
+
+-- TODO(mangoiv): assert zero legalhold devices
+
+testLHEnablePerTeam :: App ()
+testLHEnablePerTeam = do
+  startDynamicBackends [mempty] \[dom] -> do
+    -- team users
+    -- alice (team owner) and bob (member)
+    (alice, tid, [bob]) <- createTeam dom 2
+    legalholdIsEnabled tid alice `bindResponse` \resp -> do
+      resp.status `shouldMatchInt` 200
+      resp.json %. "lockStatus" `shouldMatch` "unlocked"
+      resp.json %. "status" `shouldMatch` "disabled"
+
+    withMockServer lhMockApp \lhPort _chan -> do
+      setUpLHDevice tid alice bob lhPort
+      tidStr <- asString tid
+
+      legalholdUserStatus tid alice bob `bindResponse` \resp -> do
+        resp.status `shouldMatchInt` 200
+        resp.json %. "status" `shouldMatch` "enabled"
+
+      baseRequest alice Galley Versioned (joinHttpPath ["teams", tidStr, "features", "legalhold"])
+        >>= submit "PUT"
+          . addJSONObject ["status" .= "disabled", "ttl" .= "unlimited"]
+          `bindResponse` assertLabel 403 "legalhold-whitelisted-only"
+
+      -- the put doesn't have any influence on the status being "enabled"
+      legalholdUserStatus tid alice bob `bindResponse` \resp -> do
+        resp.status `shouldMatchInt` 200
+        resp.json %. "status" `shouldMatch` "enabled"
+
+testLHGetMembersIncludesStatus :: App ()
+testLHGetMembersIncludesStatus = do
+  startDynamicBackends [mempty] \[dom] -> do
+    -- team users
+    -- alice (team owner) and bob (member)
+    (alice, tid, [bob]) <- createTeam dom 2
+
+    let statusShouldbe :: String -> App ()
+        statusShouldbe status = do
+          getTeamMembers alice tid `bindResponse` \resp -> do
+            resp.status `shouldMatchInt` 200
+            [bobMember] <-
+              resp.json %. "members" & asList >>= filterM \u -> do
+                (==) <$> asString (u %. "user") <*> objId bob
+            bobMember %. "legalhold_status" `shouldMatch` status
+
+    statusShouldbe "no_consent"
+    withMockServer lhMockApp \lhPort _chan -> do
+      statusShouldbe "no_consent"
+
+      legalholdWhitelistTeam tid alice
+        >>= assertStatus 200
+
+      -- the status messages for these have already been tested
+      postLegalHoldSettings tid alice (mkLegalHoldSettings lhPort)
+        >>= assertStatus 201
+
+      -- legalhold has been requested but is disabled
+      statusShouldbe "disabled"
+
+      requestLegalHoldDevice tid alice bob
+        >>= assertStatus 201
+
+      -- legalhold has been set to pending after requesting device
+      statusShouldbe "pending"
+
+      approveLegalHoldDevice tid bob defPassword
+        >>= assertStatus 200
+
+      -- bob has accepted the legalhold device
+      statusShouldbe "enabled"
+
+type TB s = TaggedBool s
+
+ut :: TB s -> Bool
+ut = unTagged
```

Comment:
I think I do this because someone told me at some point that they don’t like if people match on MkSomething constructors so I kinda got used to use view patterns an field accessors; will just change it though.